WordPress/Classes/Utility/Media/ImageDownloader.swift

```diff
@@ -151,14 +153,24 @@ actor ImageDownloader {
 }
 
 private final class ImageDataTask {
+    let key: String
     var subscriptions = Set<UUID>()
-    var isCancelled = false
-    var task: Task<Data, Error>
+    let task: Task<Data, Error>
+    weak var downloader: ImageDownloader?
 
-    init(subscriptions: Set<UUID> = Set<UUID>(), task: Task<Data, Error>) {
-        self.subscriptions = subscriptions
+    init(key: String, _ task: Task<Data, Error>) {
+        self.key = key
         self.task = task
     }
+
+    func getData(subscriptionID: UUID) async throws -> Data {
+        try await withTaskCancellationHandler {
```

Comment:
This is now called with no isolation, so hopefully it will skip the part of the runtime that crashes on iOS 18 betas.